src/main/java/com/powsybl/sld/server/SingleLineDiagramService.java

```diff
@@ -107,12 +112,41 @@ Pair<String, String> generateSvgAndMetadata(UUID networkUuid, String variantId,
             SingleLineDiagram.draw(network, id, svgWriter, metadataWriter, layoutParameters, compLibrary,
                     substationLayoutFactory, voltageLevelLayoutFactory, labelProvider, defaultDiagramStyleProvider, "");
 
-            return Pair.of(svgWriter.toString(), metadataWriter.toString());
+            Map<String, Object> additionalMetadata = computeAdditionalMetadata(network, id);
+
+            return SvgAndMetadata.builder()
+                    .svg(svgWriter.toString())
+                    .metadata(metadataWriter.toString())
+                    .additionalMetadata(additionalMetadata).build();
         } catch (IOException e) {
             throw new UncheckedIOException(e);
         }
     }
 
+    private Map<String, Object> computeAdditionalMetadata(Network network, String id) {
+
+        Map<String, Object> metadata = new HashMap<>();
+        metadata.put("id", id);
```

Comment:
Add static constants for metadata keys (voltageorSubstationId, voltageorSubstationName, substationCountryName) in `SvgAndMetadata` class
So all matadata names are defined centrally
Perhaps a dto `SLDInfos`

Reply:
I will do DTOs instead.